packages/ra-ui-materialui/src/list/filter/FilterLiveSearch.tsx

```diff
@@ -22,41 +23,57 @@ import TextInput from '../../input/TextInput';
  *     </Card>
  * );
  */
-const FilterLiveSearch: FC<{ source?: string }> = props => {
-    const { source = 'q', ...rest } = props;
+const FilterLiveSearch: FC<{ source?: string; label?: string }> = props => {
+    const { source = 'q', label, ...rest } = props;
+
     const { filterValues, setFilters } = useListFilterContext();
     const translate = useTranslate();
+    const formRef = useRef<any>(null);
 
+    // clear the form when the correspondind <Filter> is removed
+    if (!get(filterValues, source)) {
+        if (formRef.current) {
+            formRef.current.reset();
+        }
+    }
     const onSearchChange = (event: ChangeEvent<HTMLInputElement>) => {
         if (event.target) {
-            setFilters({ ...filterValues, [source]: event.target.value }, null);
+            const { values } = formRef.current.getState(); // get parsed value
+            const query = merge({}, filterValues, values);
+            setFilters(query, null);
         } else {
-            const { [source]: _, ...filters } = filterValues;
-            setFilters(filters, null);
+            const cleanedFilter = set(filterValues, source, null);
+            setFilters(cleanedFilter, null);
         }
     };
 
     const onSubmit = () => undefined;
 
     return (
         <Form onSubmit={onSubmit}>
-            {({ handleSubmit }) => (
-                <TextInput
-                    resettable
-                    helperText={false}
-                    source={source}
-                    label={translate('ra.action.search')}
-                    InputProps={{
-                        endAdornment: (
-                            <InputAdornment position="end">
-                                <SearchIcon color="disabled" />
-                            </InputAdornment>
-                        ),
-                    }}
-                    onChange={onSearchChange}
-                    {...rest}
-                />
-            )}
+            {({ form }) => {
+                formRef.current = form;
+                return (
+                    <TextInput
+                        resettable
+                        helperText={false}
+                        source={source}
+                        label={
```

Comment:
What do you think about setting a default value for `label` in the props declaration instead?

Reply:
@djhi, done. Cleaner.